services/types.ts

```diff
@@ -0,0 +1,146 @@
+import assert from 'assert'
+
+/**
+ * Represents a point in time
+ */
+export class Time {
```

Comment:
Could we get all these types broken out into their own files, in a new 'entities' directory? Ideally, name them after the type they represent. To still include the functionality to import all with a single import statement, add an 'index.ts' file that explicitly exports each type (syntax: export * from "./[type].ts";, or at least something similar)